### PR TITLE
[libjxl] Update to v0.11.0

### LINKED ISSUE #40970: [libjxl] update to v0.11.0
### Library name

libjxl

### New version number

v0.11.0

### Other information that may be useful (release notes, etc...)

_No response_

### PR DESCRIPTION
**Pull request by LucasChollet**
Fixes #40970

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] When updating the upstream version, the `"port-version"` is reset (removed from `vcpkg.json`).
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by LucasChollet** [comment]
@microsoft-github-policy-service agree

**Review by JonLiu1993 — APPROVED** [review]

File: ports/libjxl/portfile.cmake [hunk 1]
```diff
@@ -2,11 +2,10 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libjxl/libjxl
     REF "v${VERSION}"
-    SHA512 1d19f30c0ae94e212326a15dc128612bce207e619c55c2a020401a27d01a933de71bb35e907521dc8fd853dd5578d74d74f7a5a282f0083fd22cf9cb07ccb96e
+    SHA512 a3648a5f046cae36b3272c46525d002d490f781f44647d4d8318f0b773dd9b7902582ced5636af3489f1d6a44e3baf8ad2e5ab47d26869d16b01607d90e14053
     HEAD_REF main
     PATCHES
         fix-dependencies.patch
-        fix-tools-build.patch
```
Comment (JonLiu1993) [inline ports/libjxl/portfile.cmake:9]: The patch has been applied to the new version.

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

File: ports/libjxl/portfile.cmake (hunk 1 above)
Comment (LucasChollet) [reply ports/libjxl/portfile.cmake:9]: Sorry to not mention it, but this was indeed patched upstream in https://github.com/libjxl/libjxl/commit/a741085a1dee343f143f5fdca3212ca13d66e401

**Comment by LucasChollet** [comment]
I also tested:
 - x64-linux

**Comment by BillyONeal** [comment]
Thanks for the update and removing the no longer needed patch!

**Pull request merged**